ccpage171.py

amymhaddad opened the pull request:
There are 2 exercises in this file, separated by blank lines.

amymhaddad committed:
ccpage171.py

There are 2 exercises in this file, separated by blank lines.

amymhaddad commented on ccpage171.py:
```diff
@@ -0,0 +1,106 @@
+user.increment_login_attempts(1)
+user.increment_login_attempts(1)
+user.increment_login_attempts(1)
+user.increment_login_attempts(1)
```
@paulghaddad Is this the best way to increment by 1? Is there another way to do this automatically? I tried using a variable in the method that was equal to 0, but I got an error. Here's what I did:

```
def increment_login_attempts(self, increase_attempts):
        """Increment the login attempts"""
        attempts = 1
        self.login_attempts += attempts
```
I thought when I called the class I could just use the variable, not the number "1":
`user.increment_login_attempts(attempts)`

Because I got an error, I resorted the the longer, manual version of incrementing a number. Thoughts on this?

paulghaddad commented on ccpage171.py:
```diff
@@ -0,0 +1,106 @@
+
+class Restaurant():
+    """An attempt to model a restaurant"""
+    def __init__(self, restuaruant_name, restaurant_cuisine):
```
To add another attribute, you should pass in `number_served` to the `__init__` method. Otherwise, it will always be 5.

```
def __init__(self, restuaruant_name, restaurant_cuisine, number_served=0):
```

paulghaddad commented on ccpage171.py:
```diff
@@ -0,0 +1,106 @@
+        self.number_served = 5
+    
+    def describe_restaruant(self):
+        """Write a variable that returns a summary of the restaurant"""
```
Nice

paulghaddad commented on ccpage171.py:
```diff
@@ -0,0 +1,106 @@
+        self.number_served = guests
+
+    # Add a method to increment the total number of people served
+    def increment_number_served(self, incremented_guests):
```
Nice job here, I think `new_customer_count` may be a better variable name.

amymhaddad replied:
@paulghaddad I agree. However, the exercise problem told me to use this variable name, probably to distinguish what I'm doing here -- incrementing.

paulghaddad commented on ccpage171.py:
```diff
@@ -0,0 +1,106 @@
+    # I added a method here so I can call the method, instead of printing a statment about the total number of guests served
+    def statement_for_number_served(self):
+        """Print a statement that tells the number of people served"""
+        print(f"The total number of people served is {int(self.number_served)}.")
```
Nice job adding this method. `number_served` is already an `int`, so there's no need to cast (convert) it to a `int`.

paulghaddad commented on ccpage171.py:
```diff
@@ -0,0 +1,106 @@
+    def __init__(self, restuaruant_name, restaurant_cuisine):
+        """Initialize the attributes"""
+        self.restuaruant_name = restuaruant_name
+        self.restuaruant_cuisine = restaurant_cuisine
```
This attribute is misspelled.

paulghaddad commented on ccpage171.py:
```diff
@@ -0,0 +1,106 @@
+
+    def increment_login_attempts(self, increase_attempts):
+        """Increment the login attempts"""
+        self.login_attempts += increase_attempts
```
Since it will _always_ be increm ented by 1, there's no need to have a parameter passed in and you can just:

```
self.login_attempts += 1
```

paulghaddad commented on ccpage171.py:
```diff
@@ -0,0 +1,106 @@
+        """Increment the login attempts"""
+        self.login_attempts += increase_attempts
+    
+    def reset_login_attempts(self, reset):
```
Same thing as above...always going to reset to `0`.

paulghaddad commented on ccpage171.py:
```diff
@@ -0,0 +1,106 @@
+    
+    def print_total_login_attempts(self):
+        """Print a statement with the number of login attempts"""
+        if self.login_attempts >1: 
```
Space missing after the `>`

paulghaddad replied:
Nice job ensuring proper pluralization.

paulghaddad reviewed (changes requested):
Great work all around. Just a couple pointers on what's necessary to pass into a function (or not).

amymhaddad committed:
Update ccpage171.py

amymhaddad commented:
@paulghaddad This file has been updated with these changes:

-At 17 I added a `number_served` attribute and set it to 0.
-At 41, I removed the 'int'
-Updated the spelling of attribute in first exercise
-Update the methods on 83 and 87, respectively

amymhaddad committed:
Update ccpage171.py
